compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -925,11 +928,17 @@ class Namer { typer: Typer =>
 
     val TypeDef(name, impl @ Template(constr, _, self, _)) = original
 
-    private val (params, rest): (List[Tree], List[Tree]) = impl.body.span {
+    private val (params, restOfBody): (List[Tree], List[Tree]) = impl.body.span {
       case td: TypeDef => td.mods.is(Param)
       case vd: ValDef => vd.mods.is(ParamAccessor)
       case _ => false
     }
+    private val (restAfterParents, rest): (List[Tree], List[Tree]) =
+      if original.mods.isEnumClass then
+        val (imports :: getters :: Nil, stats): @unchecked = restOfBody.splitAt(2)
```

Comment:
It would be good to have an explanation what goes on here. Fiddling with elements like this is obscure and also fragile. There should at least be a statement that the first two elements are always `imports` (and what they are) and `getters`, and why that is so,

Reply:
I have added a comment here with more details